File: cloudbot/docs/legacy/commercial-director/assistant/agents/larisa-ivanovna/commands/get_weather.ts
import { LARISA_COMMAND_ALIASES } from "../config";
import { runWeatherWorkflow, type WeatherWorkflowDeps } from "../workflows/weather.workflow";

export function createGetWeatherCommand(deps: WeatherWorkflowDeps) {
  return {
    name: "get_weather",
    aliases: LARISA_COMMAND_ALIASES.getWeather,
    async execute(input: { dateMsk: string; city?: string }) {
      const result = await runWeatherWorkflow(input, deps);

      return {
        text: result.text,
        payload: result,
      };
    },
  };
}

File: cloudbot/apps/larisa_ivanovna/commands/get_weather.py
"""Команда погодного блока."""

from __future__ import annotations

from ..config import COMMAND_ALIASES, DEFAULT_CONFIG
from ..workflows.weather import WeatherWorkflowDeps, run_weather_workflow


def build_command(deps: WeatherWorkflowDeps) -> dict[str, object]:
    def handler(payload: dict[str, str], context: dict[str, object] | None = None) -> dict[str, object]:
        result = run_weather_workflow(
            date_msk=str(payload.get("date_msk") or ""),
            city=str(payload.get("city") or DEFAULT_CONFIG.default_city),
            deps=deps,
        )
        return {
            "text": str(result.get("text") or ""),
            "payload": result,
        }

    return {
        "name": "get_weather",
        "aliases": COMMAND_ALIASES["get_weather"],
        "handler": handler,
    }
